crates/llm-ls/src/retrieval.rs

```diff
@@ -459,31 +480,63 @@ impl SnippetRetriever {
             let window = lines[start_line..end_line].to_vec();
             let snippet = window.join("\n");
             if snippet.is_empty() {
-                continue;
+                    debug!("snippet {file_url}[{start_line}, {end_line}] empty");
+                    continue;
             }
+            snippets.push(Snippet{ file_url: file_url.clone().into(), code: snippet, start_line, end_line });
+        }
+        {
+            let nb_snippets = snippets.len();
+            let steps = self.window_step;
+            debug!("Found {nb_snippets} snippets in {file_url}: {start}, {end}, {steps}");
+        }
 
-            let mut encoding = self.tokenizer.encode(snippet.clone(), true)?;
-            encoding.truncate(512, 1, TruncationDirection::Right);
-            let result = self.generate_embedding(encoding, self.model.clone()).await;
-            let embedding = match result {
-                Ok(e) => e,
+        // Group by length to reduce padding effect
+        snippets.sort_by(|first, second| first.code.len().cmp(&second.code.len()));
```

Comment:
Vec::sort_by is O(n * log(n)) at worst, and given the side of the window, most code files contains around 10-20 snippets, I have not measured but it should pretty fast, I can add a spawn_blocking in case there is some very large files. I saw that sort_unstable_by may be faster because there is no allocation so this could also be an improvement

Reply:
Let's go with unstable for now then, we can always check later how long this takes for big files.